### DISCUSSION TITLE
Wishlists and issues to do next!

### DISCUSSION THREAD

**Discussion by kimkulling** [assimp/assimp#5869] [Ideas]
If there is any feature which you need next or you need to ping me: use this topic.

**Comment by kcrif** [comment]
Improved performance for Asset::FindUniqueID, see #3444, Should be a simple fix

**Comment by kimkulling** [comment]
Good point!

**Comment by kimkulling** [comment]
What the ... . This looks terribly slow.

**Comment by johnb003** [comment]
Pick formats that matter most, USD, FBX, glb/gltf mostly, and extend core functionality for these formats with a matrix of feature falloff on other formats, but don't let the other formats limit productivity where it matters most.

The problem is that when the library is first considered for a project it's great that it has a wide range of supported formats, and it seems like people have gotten it working, but then you quickly run into limitations, and fixing this library, if the key value is wide support, feels daunting because so many of the formats aren't that important.

There are a number of basic issues for me that are pushing me to consider direct SDK alternatives for the formats I care most about.

* I don't like that the skeleton is not identifiable in the scene without skinned meshes.
* I saw instances when preserve pivots is used, it inserted extra transforms, but they weren't added to the scene, so you're left with trying to parse the strings to understand where in the scene those channels are related.
* Lack of tangents limits usefulness of curves that aren't baked.
* Pre/post infinity options aren't as complete as in the dcc tools (for example cycle w/ offset, I think?)
* Attribute animation is a big one -- I need general float anims, and the lib assumes all the animation is just for transforms.
* blendshapes would be nice to load, and I'm talking about sections of mesh being sculpted in static poses, where you'd drive those activations with animated weights (not vertex animation).

In general the way the skeleton is determined by gathering what are joints out of skin clusters or guessing based on what's animated, and then crawling the hierarchy, feels like a lot of unnecessary processing that makes the library less easy to use.

I guess my expectation is:
Don't assume the project maturity is good enough to just polish bugs and performance and forget about features. There's still a lot improvements needed for the main architectural design of it and key features that will make or break the success in the long run.

**Comment by EugenyP** [comment]
Support for the MDS format from Return to Castle Wolfenstein

https://mino-git.github.io/rtcw-wet-blender-model-tools/publications/MDSFileFormat.html
https://mino-git.github.io/rtcw-wet-blender-model-tools/mds.html

**Comment by tellypresence** [comment]
@EugenyP 
Raised new issue #6596 for MDS support request (better to track these sorts of requests as issues) -- please feel free to comment and add detail there

**Comment by 11mleuthold** [comment]
Support for [OpenCTM](https://en.wikipedia.org/wiki/OpenCTM) CTM is a highly efficient mesh compressing algorithm, written in C ([Github](https://github.com/Danny02/OpenCTM/))

**Reply by tellypresence** [reply]
@11mleuthold Added OpenCTM to list of requested formats (#6097), suggest raising an issue so progress can be tracked
